aiida_testing/mock_code/_env_keys.py

```diff
@@ -2,20 +2,96 @@
 """
 Defines the environment variable names for the mock code execution.
 """
-
+from dataclasses import dataclass
 from enum import Enum
+import inspect
+import os
+from pathlib import Path
+import typing as ty
+
+from ._hasher import InputHasher, load_hasher
 
 
-class EnvKeys(Enum):
+@dataclass
+class MockVariables:  # pylint: disable=too-many-instance-attributes
+    """
+    A class containing variables defined for the mock code execution.
+    """
+
+    log_file: Path
+    label: str
+    data_dir: Path
+    executable_path: str
+    ignore_files: ty.Iterable[str]
+    ignore_paths: ty.Iterable[str]
+    regenerate_data: bool
+    fail_on_missing: bool
+    _hasher: ty.Union[str, ty.Type[InputHasher]]
+
+    @classmethod
+    def from_env(cls) -> "MockVariables":
+        """
+        Create a MockVariables instance from the environment variables.
+        """
+        return cls(
+            log_file=Path(os.environ[_EnvKeys.LOG_FILE.value]),
+            label=os.environ[_EnvKeys.LABEL.value],
+            data_dir=Path(os.environ[_EnvKeys.DATA_DIR.value]),
+            executable_path=os.environ[_EnvKeys.EXECUTABLE_PATH.value],
+            ignore_files=os.environ[_EnvKeys.IGNORE_FILES.value].split(":"),
+            ignore_paths=os.environ[_EnvKeys.IGNORE_PATHS.value].split(":"),
+            regenerate_data=os.environ[_EnvKeys.REGENERATE_DATA.value] == "True",
+            fail_on_missing=os.environ[_EnvKeys.FAIL_ON_MISSING.value] == "True",
+            _hasher=os.environ.get(_EnvKeys.HASHER.value, InputHasher),
+        )
+
+    def get_hasher(self) -> ty.Type[InputHasher]:
+        """
+        Return the hasher class.
+        """
+        if isinstance(self._hasher, str):
+            # split the string into the file path and the class name
+            file_path, class_name = self._hasher.rsplit("::", 1)
+            return load_hasher(file_path, class_name)
+        return self._hasher
+
+    def to_env(self) -> str:
+        """
+        Return a string that can be used to export the environmental variables
+        """
+        string = inspect.cleandoc(
+            f"""
+                export {_EnvKeys.LOG_FILE.value}="{self.log_file}"
```

Comment:
I guess people won't really look at the generated file but that's quite a bit of indentation here...

Reply:
Oh the cleandoc function it is wrapped in removes all that indentation

Reply:
Interesting that the stdlib has two functions that do almost the same.. I've always used [`textwrap.dedent`](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) for this.